src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [] ;
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(obj) {
+    if (obj instanceof Herbivore) {
+      if (obj.hidden === false) {
+        obj.health -= 50;
+
+        if (obj.health <= 0) {
+          Animal.alive = Animal.alive.filter(animal => animal !== obj);
+        }
```

Comment:
too much unnecessary nesting.
also, don't check a boolean value for equality with another boolean value. simply use `!`